src/container/RelayRenderer.js

```diff
@@ -253,29 +212,7 @@ class RelayRenderer extends React.Component {
    * @private
    */
   _shouldUpdate(): boolean {
-    const {activeContainer, activeEnvironment, activeQueryConfig} = this.state;
-    const {Container, queryConfig, environment} = this.props;
-    return (
-      (!activeContainer || Container === activeContainer) &&
-      (!activeEnvironment || environment === activeEnvironment) &&
-      (!activeQueryConfig || queryConfig === activeQueryConfig)
-    );
-  }
-
-  /**
-   * @private
-   */
-  _runQueriesAndSetState(props: RelayRendererProps): void {
-    this._runQueries(props);
-    this.setState(
-      this._buildState(
-        this.state.activeContainer,
-        this.state.activeEnvironment,
-        this.state.activeQueryConfig,
-        null,
-        null
-      )
-    );
+    return !!this.state.readyState || !this.state.active;
```

Comment:
After receiving a `readyState` the first time, we set `active` to `true` and never change back. So, with just `return !this.state.active;` we would never update after that.